### PR TITLE
Cleanup, docu and refactoring in ilCtrl

### PR DESCRIPTION
**Pull request by klees**
This is some cleanup, documentation and refactoring in ilCtrl. I do not intend to change any functionality. Changes are:

* lots of docu, I hope I understood everything well
* added access qualifiers
* removal of some deprecated functionality
* removal of some code that does not cause any effect
* removal of some code that was outcommented
* some renaming and refactoring in the whole "class-id"-shebang

I also think, that this class is a good target for tests and refactoring, as it is very central in the system and should be married with the new HTTPService and the UI-Framework at some point. There seem to be multiple classes hidden in ilCtrl regarding the responsibilities the code has. The implementation of the caching by using a singleton could be replaced by an injected dependency.

### CODE REVIEW

File: Services/UICore/classes/class.ilCtrl.php [hunk 1]
```diff
@@ -983,13 +876,15 @@ private function getPathNew($a_source_node, $a_target_node)
...
 		return $path;
 	}
 
 	/**
-	 * set target script name
+	 * Set target script name.
+	 *
+	 * This is in fact used in various places, but as `grep -r setTargetScript .`
+	 * reveals is always set to "ilias.php". The target script stuff seems to be
+	 * pointless.
```
Comment (alex40724) [inline Services/UICore/classes/class.ilCtrl.php]: Then we should get rid of it. In older ILIAS versions there have been several scripts like repository.php, admin.php, ... It was a long time goal to get rid of these and use ilias.php only.

Comment (klees) [reply Services/UICore/classes/class.ilCtrl.php]: I was not sure if there is some non-obvious reason for its existence. Will remove...

**Comment by klees** [comment]
Hi @alex40724, could you give me some feedback how we could go on with this? Shall I go on keeping this mergeable?

**Comment by Amstutz** [comment]
Can this be merged soon? Would be bad if it has to wait one full release again.

**Comment by klees** [comment]
Merged current trunk.

File: Modules/Bibliographic/classes/class.ilObjBibliographicGUI.php [hunk 2]
```diff
@@ -231,7 +231,6 @@ public static function _goto($a_target) {
 		global $DIC;
 
 		$id = explode("_", $a_target);
-		$DIC->ctrl()->setTargetScript("ilias.php");
```
Comment (chfsx) [inline Modules/Bibliographic/classes/class.ilObjBibliographicGUI.php]: LGTM

File: Modules/OrgUnit/classes/class.ilObjOrgUnitGUI.php [hunk 3]
```diff
@@ -640,7 +640,6 @@ public static function _goto($ref_id) {
 		global $DIC;
 		$ilCtrl = $DIC['ilCtrl'];
 		$ilCtrl->initBaseClass("ilAdministrationGUI");
-		$ilCtrl->setTargetScript("ilias.php");
```
Comment (chfsx) [inline Modules/OrgUnit/classes/class.ilObjOrgUnitGUI.php:644]: LGTM

File: Services/Component/classes/class.ilPlugin.php [hunk 4]
```diff
@@ -966,7 +966,6 @@ function update()
 		$structure_reader = new ilCtrlStructureReader();
 		$structure_reader->readStructure(true, "./".$this->getDirectory(), $this->getPrefix(),
 			$this->getDirectory());
-		//		$ilCtrl->storeCommonStructures();
```
Comment (chfsx) [inline Services/Component/classes/class.ilPlugin.php:969]: Thanks!

File: Services/MainMenu/classes/class.ilMainMenuGUI.php [hunk 5]
```diff
@@ -1065,14 +1065,10 @@ function renderHelpButtons()
 			$this->tpl->parseCurrentBlock();
 
 			// always set ajax url
-			$ts = $ilCtrl->getTargetScript();
```
Comment (chfsx) [inline Services/MainMenu/classes/class.ilMainMenuGUI.php:1068]: Can't decide whether this is needed or not.

File: Services/Repository/classes/class.ilObjectPluginGUI.php [hunk 6]
```diff
@@ -382,14 +382,8 @@ function afterSave(ilObject $newObj)
 		ilUtil::sendSuccess($this->lng->txt("object_added"),true);
 
 		$ilCtrl->initBaseClass("ilObjPluginDispatchGUI");
-		$ilCtrl->setTargetScript("ilias.php");
 		$ilCtrl->getCallStructure(strtolower("ilObjPluginDispatchGUI"));
 		
-//var_dump($ilCtrl->call_node);
```
Comment (chfsx) [inline Services/Repository/classes/class.ilObjectPluginGUI.php:388]: Thanks!

**Review by chfsx — APPROVED** [review]
LG for my parts. Thanks!

**Comment by alex40724** [comment]
Created a rebased version without conflicts at
https://github.com/ILIAS-eLearning/ILIAS/pull/1289

**Comment by alex40724** [comment]
Code is merged to trunk now, see
https://github.com/ILIAS-eLearning/ILIAS/commit/0fce550ab13b38161f8117e1fab283aae15de5c8

**Pull request closed without merging**